### PR TITLE
add leader elect args

### LINKED ISSUE #19: Increase number of replica of the LWS controller to be deployed to cluster for better failover
Right now, we only deploy one replica of LWS controller in the cluster. When the controller is down, the reconciliation will not work and need to wait for it becoming ready again. We need to increase replica count and enabled leader election

### PR DESCRIPTION
**Pull request by gujingit**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes
Fixes #19 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -52,18 +53,47 @@ func init() {
...
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
 	flag.Float64Var(&qps, "kube-api-qps", 500, "Maximum QPS to use while talking with Kubernetes API")
 	flag.IntVar(&burst, "kube-api-burst", 500, "Maximum burst for throttle while talking with Kubernetes API")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", true,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.DurationVar(&leaderElectLeaseDuration, "leader-elect-lease-duration", 15*time.Second,
+		"he duration that non-leader candidates will wait after observing a leadership renewal until attempting to acquire "+
```
Comment (liurupeng) [inline cmd/main.go:80]:
```suggestion
		"The duration that non-leader candidates will wait after observing a leadership renewal until attempting to acquire "+
```

**Comment by liurupeng** [comment]
Were you able to test manually to verify the leader election functionality work as expected? @gujingit

**Comment by gujingit** [comment]
> Were you able to test manually to verify the leader election functionality work as expected? @gujingit
### Test 1: deploy lws-controller-manager with 2 replicas

1. Deploy lws-controller-manager with 2 replicas
```
NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
lws-controller-manager   2/2     2            2           28m
```
2. Deploy a simple lws application
```
kubectl apply -f docs/examples/sample/lws.yaml 
```
outputs
```
NAME                         READY   STATUS    RESTARTS   AGE
leaderworkerset-sample-0     1/1     Running   0          9s
leaderworkerset-sample-0-1   1/1     Running   0          9s
leaderworkerset-sample-0-2   1/1     Running   0          9s
leaderworkerset-sample-0-3   1/1     Running   0          9s
leaderworkerset-sample-1     1/1     Running   0          9s
leaderworkerset-sample-1-1   1/1     Running   0          9s
leaderworkerset-sample-1-2   1/1     Running   0          9s
leaderworkerset-sample-1-3   1/1     Running   0          9s
leaderworkerset-sample-2     1/1     Running   0          9s
leaderworkerset-sample-2-1   1/1     Running   0          9s
leaderworkerset-sample-2-2   1/1     Running   0          9s
leaderworkerset-sample-2-3   1/1     Running   0          9s
```

### Test 2:  Leader election lost
1. Get the current leader
```
NAME                                      READY   STATUS    RESTARTS   AGE
lws-controller-manager-77489f74b8-74mwv   2/2     Running   0          8m19s
lws-controller-manager-77489f74b8-qs288   2/2     Running   0          8m40s
```
leader
```
kubectl -n lws-system get lease b8b2488c.x-k8s.io
####
NAME                HOLDER                                                                         AGE
b8b2488c.x-k8s.io   lws-controller-manager-77489f74b8-qs288_7f45752d-2de4-4da8-ab31-7313dc04f1ff   29m
```
2. Kill the leader pod
```
kubectl  -n lws-system delete po lws-controller-manager-77489f74b8-qs288
```
The leader was changed to the other pod automatically
```
kubectl -n lws-system get lease b8b2488c.x-k8s.io
####
NAME                HOLDER                                                                         AGE
b8b2488c.x-k8s.io   lws-controller-manager-77489f74b8-74mwv_2d1b2863-e787-42e6-af13-71eb2a9aae9f   31m
```

**Review by kerthcet — COMMENTED** [review]
/hold

Should we make this the default behavior?

**Comment by liurupeng** [comment]
> /hold
> 
> Should we make this the default behavior?

I think so, but do you know what might be the downside for that?

**Comment by kerthcet** [comment]
Nothing special
/hold cancel

**Comment by liurupeng** [comment]
/lgtm
/approve

Thanks @gujingit

**Pull request merged**